src/main/java/org/folio/gobi/Mapper.java

```diff
@@ -91,32 +92,49 @@ public Mapper(Map<Mapping.Field, DataSource> mappings) {
             ids.add(productId);
             detail.setProductIds(ids);
             
+            if(!isObjectEmpty(adjustment)) 
```

Comment:
All of these if statements is going to bloat the code complexity score...  you might consider doing something like creating methods for this...  e.g. `setDetailsifPresent(detail)`

Reply:
so you mean the each of those set methods will just have the if statement?

Reply:
```
private void setIfPresent(Object o, Consumer<Object> setter) {
  if(!isObjectEmpty(o)) {
    setter.accept(o);
  }    
}
```

Then you can do...

```
setIfPresent(cost, o -> pol.setDetails((Cost)o));
setIfPresent(detail, o -> pol.setDetails((Details)o));
...

```

Reply:
and ...

```
setIfPresent(reportingCode, o -> {
  List<ReportingCode> reportingCodes = new ArrayList<>();
  reportingCodes.add((ReportingCode)o);
  pol.setReportingCodes(reportingCodes);
});
```